sorted_list_dict.py

```diff
@@ -12,20 +12,36 @@ def __eq__(self, other):
     def __str__(self) -> str:
         return self.__list.__str__()
 
+    def __len__(self) -> int:
+        return self.__list.length()
+
     def __find_gte_or_length(self, key: str) -> int:
         '''
         gte means greater or equal
         '''
-        # TODO: binary search
+        # binary search
+
+        left = 0  # left border
+        right = len(self.__list) - 1  # right border
+        middle = 0
 
-        index = 0
+        while left <= right:
```

Comment:
In the current context we have:
`[]` - empty: `0 <= -1`, `False`,
`[('aaa', 10)]` - with one element: `0 <= 0`, `True`,
`[('aaa', 10), ('bbb', 20), ...]` - more than 1 element: `0 <= len`, `True`.